pkg/query-service/auth/auth.go

```diff
@@ -375,3 +428,35 @@ func passwordMatch(hash, password string) bool {
 	}
 	return true
 }
+
+func GenerateJWTForUser(user *model.User) (model.UserJwtObject, error) {
+	j := model.UserJwtObject{}
+	var err error
+	j.AccessJwtExpiry = time.Now().Add(JwtExpiry).Unix()
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"id":    user.Id,
+		"gid":   user.GroupId,
+		"email": user.Email,
+		"exp":   j.AccessJwtExpiry,
+	})
+
+	j.AccessJwt, err = token.SignedString([]byte(JwtSecret))
+	if err != nil {
+		return j, errors.Errorf("failed to encode jwt: %v", err)
+	}
+
+	j.RefreshJwtExpiry = time.Now().Add(JwtRefresh).Unix()
+	token = jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"id":    user.Id,
+		"gid":   user.GroupId,
+		"email": user.Email,
+		"exp":   j.RefreshJwtExpiry,
+	})
+
+	j.RefreshJwt, err = token.SignedString([]byte(JwtSecret))
+	if err != nil {
+		return j, errors.Errorf("failed to encode jwt: %v", err)
```

Comment:
```suggestion
		return nil, errors.Errorf("failed to encode jwt: %v", err)
```

Why return the object when there is an error?

Reply:
The return var is not pointer in this case. And this is existing code, I wouldn't want to change in this release.